common/ledger/blockledger/fileledger/factory_test.go

```diff
@@ -7,74 +7,83 @@ SPDX-License-Identifier: Apache-2.0
 package fileledger
 
 import (
-	"fmt"
+	"errors"
 	"io/ioutil"
 	"os"
 	"testing"
 
-	"github.com/hyperledger/fabric/common/ledger/blkstorage"
 	"github.com/hyperledger/fabric/common/ledger/blockledger"
+	"github.com/hyperledger/fabric/common/ledger/blockledger/fileledger/mock"
 	"github.com/hyperledger/fabric/common/metrics/disabled"
 	"github.com/stretchr/testify/require"
 )
 
-type mockBlockStoreProvider struct {
-	blockstore *blkstorage.BlockStore
-	list       []string
-	error      error
-}
-
-func (mbsp *mockBlockStoreProvider) Open(ledgerid string) (*blkstorage.BlockStore, error) {
-	return mbsp.blockstore, mbsp.error
-}
-
-func (mbsp *mockBlockStoreProvider) List() ([]string, error) {
-	return mbsp.list, mbsp.error
-}
+func TestBlockStoreProviderErrors(t *testing.T) {
+	mockBlockStore := &mock.BlockStoreProvider{}
```

Comment:
Nit. It is possible to directly use the block store rather than using a mock but not related to this PR.